src/Chapter3.hs

```diff
@@ -907,9 +998,23 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold x) (Gold y) = Gold (x + y)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append x y = x ++ y
+
+instance Append (Maybe [a]) where
```

Comment:
This instance implements only `Append` typeclass for `Maybe [a]`, but it won't work for `Maybe Gold` for example. There's a way to make this instance more generic :slightly_smiling_face: